apps/ios/GuideDogs/Code/App/Settings/SettingsContext.swift

```diff
@@ -380,6 +380,8 @@ class SettingsContext {
         }
         set {
             userDefaults.set(newValue, forKey: Keys.leaveImmediateVicinityDistance)
+            // Ensure leave is always 10m greater than enter
+            userDefaults.set(newValue - 15.0, forKey: Keys.enterImmediateVicinityDistance)
```

Comment:
```suggestion
            userDefaults.set(max(newValue - 15.0, 0.0), forKey: Keys.enterImmediateVicinityDistance)
```
There's probably no way this could end up negative with the current code, but it might be a good idea to make sure.

Reply:
Agreed. Also, this is only ever used to compare against some other nonnegative distance, so even if it does turn negative, the behavior would be the same as if it were zero.